dev_tools/requirements/re-pip-compile-in-docker.sh

```diff
@@ -0,0 +1,35 @@
+#!/bin/bash
+
+#  Copyright 2023 Google LLC
+#
+#  Licensed under the Apache License, Version 2.0 (the "License");
+#  you may not use this file except in compliance with the License.
+#  You may obtain a copy of the License at
+#
+#      https://www.apache.org/licenses/LICENSE-2.0
+#
+#  Unless required by applicable law or agreed to in writing, software
+#  distributed under the License is distributed on an "AS IS" BASIS,
+#  WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+#  See the License for the specific language governing permissions and
+#  limitations under the License.
+
+set -ex
+
+# https://stackoverflow.com/q/59895
+SCRIPT_DIR="$(cd -- "$(dirname -- "${BASH_SOURCE[0]}")" &>/dev/null && pwd)"
+readonly SCRIPT_DIR
+
+# We use docker to run the actual `pip-compile` command because its
```

Comment:
What is the workflow for regenerating new requirement files now? Should we run this script on our local machine and add?

Reply:
If you run `re-pip-compile-in-docker.sh` on a workstation with docker installed, it should just work :tm:

You can also run the python script outside of docker but its output may vary depending on the environment you run it from